src/main/scala/com/fulcrumgenomics/bam/CollectErccMetrics.scala

```diff
@@ -0,0 +1,253 @@
+      }
+      .foreach { r =>
+        erccFragments += 1
+        erccFragmentCounter.count(r.refName)
+      }
+    in.safelyClose()
+
+    // Transform to log2 counts to compute some simple correlation and linear regression metrics.
+    val counts         = ListBuffer[Double]()
+    val concentrations = ListBuffer[Double]()
+    erccData.iterator.foreach { case (name, concentration) =>
+      val count = erccFragmentCounter.countOf(name)
+      if (minTranscriptCount <= count) {
+        counts += log2(count)
+        concentrations += log2(concentration)
+      }
+    }
+
+    // Rocket science
+    val simpleRegression = new SimpleRegression()
+    counts.zip(concentrations).foreach { case (cnt, conc) => simpleRegression.addData(cnt, conc) }
+
+    val erccMetrics = ErccSummaryMetrics(
+      total_reads                = totalReads,
+      ercc_reads                 = erccReads,
+      pct_ercc_reads             = if (totalReads == 0) 0 else 100.0 * erccReads / totalReads.toDouble,
+      ercc_fragments             = erccFragments,
+      total_transcripts          = erccFragmentCounter.iterator.length,
+      total_filtered_transcripts = erccFragmentCounter.iterator.count(_._2 >= minTranscriptCount),
+      pearsons_correlation       = zeroIfNaNOrInsufficientData { new PearsonsCorrelation().correlation(counts.toArray, concentrations.toArray) },
+      spearmans_correlation      = zeroIfNaNOrInsufficientData { new SpearmansCorrelation().correlation(counts.toArray, concentrations.toArray) },
+      slope                      = zeroIfNaN { simpleRegression.getSlope },
+      r_square                   = zeroIfNaN { simpleRegression.getRSquare },
+      intercept                  = zeroIfNaN { simpleRegression.getIntercept }
+    )
+
+    // Write the output
+    {
+      def f(ext: String): FilePath = PathUtil.pathTo(output + ext)
+
+      val summaryPath  = f(".ercc_summary_metrics.txt")
+      val detailedPath = f(".ercc_detailed_metrics.txt")
+      val plotPath     = f(".ercc_plot.pdf")
+
+      // summary metrics file
+      Metric.write[ErccSummaryMetrics](summaryPath, erccMetrics)
+
+      // detailed (per-ERCC-transcript) file
+      val metrics = in.header.getSequenceDictionary
+        .getSequences
+        .map(_.getSequenceName)
+        .filter { name => erccData.contains(name) }
+        .map { name =>
+          val count         = erccFragmentCounter.countOf(name)
+          val concentration = erccData(name)
+          ErccDetailedMetric(name=name, concentration=concentration, count=count)
+        }.toSeq
+      Metric.write[ErccDetailedMetric](detailedPath, metrics)
+
+      // And try plotting
+      if (metrics.isEmpty || totalReads == 0 || simpleRegression.getN <= 1) {
+        logger.warning("No metrics generated (is your BAM empty?). Plots will not be generated.")
+      }
+      else {
+        val name = PathUtil.basename(input, trimExt=true).toString
```

Comment:
Or maybe re-use `output` here?  I think elsewhere I've tended to pull the sample or library name from the header.